### DISCUSSION TITLE
How is ReactiveViewModelMappings generated

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#10214] [Q&A]
I'm analyzing this sample to use as a starting block for my own application. There is one part that I can't figure out, how is the the ReactiveViewModelMappings file generated? I can see the generated file, but when I copy the Todo solution this file is not generated any more. I've tried to search for source generators, etc, and I can see that there is a reference to the Uno.SourceGenerationTasks nuget package, but I can't find any other references to how this file is generated. Can someone please fill me in how this file is generated? It's referenced in this line:
https://github.com/unoplatform/uno.todo/blob/main/src/ToDo.UI/App.xaml.host.cs#L64

**Comment by dr1rrb** [comment]
The file is generated by the `FeedsGenerator` in the `Uno.Extensions.Reactive` package.

To get it generated you only need to install that package and have a class named `<Something>ViewModel`. It will contains a mapping between the VM you declared and the class that has been generated from it which is more "bindable friendly" (public methods are converted to `ICommand` and `IFeed` properties are exposed as basic properties).

If you are not planning to use feeds, you can just remove the `ReactiveViewModelMappings.ViewModelMappings` argument on your referenced line. 

About feeds: https://platform.uno/docs/articles/external/uno.extensions/doc/Overview/Reactive/concept.html

**Reply by HakanL** [reply]
Thanks @dr1rrb for that information and clarification. So how would I troubleshoot this, in my copy of ToDo (I've copied most of it and then replaced the Platform projects with a WinUI and Linux FB variant) everything compiles, except that it can't find this generated class.
Also thanks for the link about Feeds, I'm not sure if I will use that, I just wanted to have a good starting point in my own project.

**Reply by HakanL** [reply]
I did upgrade all the Uno dependencies to latest version 2.2.0, and in the Todo app it was 2.1.0-dev.184, is it possible something changed in 2.2 on how it finds the view models?

**Reply by dr1rrb** [reply]
No, nothing should have changed regarding VM resolution since few months. 

I just realized that your VM class must also be `partial`.

Otherwise, are you able to share the code with us? I'm not sure I understand your setup.

**Reply by HakanL** [reply]
It's a private repo on Bitbucket, but I'm happy to share it with you, I just can't make it public (once it's set for private I can't change it). It doesn't contain anything sensitive. Do you have a user on Bitbucket and I can invite you?

**Reply by dr1rrb** [reply]
Yes I do, my username is dr1rrb. I think that you will need my account's email, but I rather not make it public here. So you can send an email to info@platform.uno (not an invitation: there is no bitbucket account associated to that email), then I'll be able to send you my bitbucket account email (quite complex process :P )

**Reply by HakanL** [reply]
Thanks, yes I think I need an email. I sent you an email and let's go from there. Really appreciate the help!
